HabitTracker/Program.cs

```diff
@@ -0,0 +1,292 @@
+        {
+            Console.WriteLine("Invalid input. Please enter a valid option.");
+            Console.Write("Option: ");
+            input = Console.ReadLine();
+        }
+
+        switch (option)
+        {
+            case 0:
+                Console.WriteLine("This application is now closing. Goodbye!");
+                closeApp = true;
+                break;
+            case 1:
+                ViewAllRecords();
+                break;
+            case 2:
+                InsertRecord();
+                break;
+            case 3:
+                UpdateRecord();
+                break;
+            case 4:
+                DeleteRecord();
+                break;
+        }
+    }
+}
+
+void ViewAllRecords()
+{
+    using (var connection = new SqliteConnection(connectionString))
+    {
+        connection.Open();
+        var tableCmd = connection.CreateCommand();
+
+        tableCmd.CommandText =
+            $"SELECT * FROM pet_the_dog";
+
+        List<DogPets> tableData = new();
+
+        SqliteDataReader reader = tableCmd.ExecuteReader();
+
+        if (reader.HasRows)
+        {
+            while (reader.Read()) // returns a bool value. True if more rows, false if none
+            {
+                tableData.Add(new DogPets
+                {
+                    Id = reader.GetInt32(0),
+                    Date = DateTime.ParseExact(reader.GetString(1), "yy-MM-dd",
+                        new CultureInfo(
+                            "en-Gb")), // Parses string from Sqlite to DateTime format. Couldn't get it to go in format yy-MM-dd
+                    Quantity = reader.GetInt32(2)
+                });
+            }
+
+            connection.Close();
+
+            Console.WriteLine("\n==========Dog Petting Tracker==========");
+            foreach (var row in tableData)
+            {
+                Console.WriteLine(
+                    $"{row.Id} - {row.Date: yyyy-MMM-dd} - No. of Dog Pets: {row.Quantity}"); // Was able to parse DateTime to new format here
+            }
+
+            Console.WriteLine("=======================================");
+        }
+        else
+        {
+            Console.WriteLine("\nNo rows in table to display.");
+            Console.WriteLine("=======================================");
+        }
+    }
+}
+
+void InsertRecord()
+{
+    Console.Clear();
+
+    var date = GetDateInput(
+        "\nPlease enter a date for the record (format: yy-mm-dd). Type 0 to return to the main menu: ");
+    var quantity =
+        GetNumberInput(
+            "Please enter the number of times you pet the dog on this date or type 0 to return to the menu: ");
+
+    using var connection = new SqliteConnection(connectionString);
+    connection.Open();
+    var tableCmd = connection.CreateCommand();
+
+    tableCmd.CommandText =
+        $"INSERT INTO pet_the_dog (Date, Quantity) VALUES ('{date}', {quantity})";
+
+    tableCmd.ExecuteNonQuery();
+    connection.Close();
```

Comment:
The using statement will take care of [closing the connection](https://learn.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlconnection.close?view=dotnet-plat-ext-8.0#examples).